packages/devtools_app/lib/src/screens/memory/panes/diff/controller/heap_diff.dart

```diff
@@ -17,10 +20,34 @@ class HeapDiffStore {
 
   DiffHeapClasses compare(AdaptedHeap heap1, AdaptedHeap heap2) {
     final couple = _HeapCouple(heap1, heap2);
-    return _store.putIfAbsent(couple, () => DiffHeapClasses(couple));
+    return _store.putIfAbsent(couple, () => _calculateDiffGaWrapper(couple));
   }
 }
 
+DiffHeapClasses _calculateDiffGaWrapper(_HeapCouple couple) {
+  late final DiffHeapClasses result;
+  ga.timeSync(
+    analytics_constants.memory,
+    analytics_constants.MemoryTime.calculateDiff,
+    syncOperation: () => result = DiffHeapClasses(couple),
+    screenMetricsProvider: () => _DiffAnalyticsMetrics(
+      objectsBefore: couple.older.data.objects.length,
+      objectsAfter: couple.younger.data.objects.length,
+    ),
+  );
+  return result;
```

Comment:
if the catch block in timeSync is hit, this will throw a late initialization error. Should we let result be nullable to handle this? or rethrow?

Reply:
I'm not referring to ProcessCancelledException. If _any_ exception is thrown by the function passed to syncOperation, the catch block will be hit.